policyengine_us/variables/gov/states/nm/tax/income/deductions/nm_itemized_deductions.py

```diff
@@ -0,0 +1,46 @@
+from policyengine_us.model_api import *
+
+
+class nm_itemized_deductions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "New Mexico itemized deductions"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf#page=28",
+    )
+    defined_for = StateCode.NM
+
+    def formula(tax_unit, period, parameters):
+        itemizes = tax_unit("tax_unit_itemizes", period)
+        filing_status = tax_unit("filing_status", period)
+
+        p = parameters(period).gov.irs.deductions
+        # tax/income, federal Schedule A, line 5a.
+        slat_sales_or_income = tax_unit(
+            "state_and_local_sales_or_income_tax", period
+        )
+        # state_and_local_tax, line 5d
+        total_salt = (
+            add(tax_unit, period, ["real_estate_taxes"]) + slat_sales_or_income
+        )
+
+        # ratio = round(slat_sales_or_income[0] / total_salt[0], 4)
+
+        salt_ratio = np.zeros_like(total_salt)
+        mask = total_salt != 0
+        salt_ratio[mask] = slat_sales_or_income[mask] / total_salt[mask]
+
+        salt_cap = p.itemized.salt_and_real_estate.cap[filing_status]
+        # salt_claimed = tax_unit("salt_deduction", period)
+
+        salt = salt_ratio * min_(salt_cap, total_salt)
```

Comment:
dont we do the multiplication first and the we take the smaller amount (Line 5 and 6)